### PR TITLE
chore: separate publish installer process

### PR DESCRIPTION
**Pull request by yihau**
#### Problem

Due to #26064, I found two things:

1. We upload installer script many times although they are the same
2. When we try to upload something, We run s3cmd in docker and volume the target files into it. The bug is that our x86 mac seems cache volume file for a while. I can reproduce the bug but haven't found the real reason.

I think we should solve both but ideally when this PR been merged and then 2.'s bug will be hidden. I'll merge this first and still investigating 2.

#### Summary of Changes

1. reorder steps attributes
4. extract publish installer logic as an independent step

### CODE REVIEW

File: ci/publish-installer.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+set -e
+
+cd "$(dirname "$0")/.."
+
+# check does it need to publish
+if [[ -n $DO_NOT_PUBLISH_TAR ]]; then
+	echo "Skipping publishing install wrapper"
```
Comment (mvines) [inline ci/publish-installer.sh:8]: tabs 💔

**Review by mvines — APPROVED** [review]
lgtm, thanks.

please backport to both v1.10 and v.1.9

File: ci/publish-installer.sh (hunk 1 above)
Comment (yihau) [reply ci/publish-installer.sh]: modified them into spaces:2 😈

**Comment by yihau** [comment]
this CI modification is hidden in the secondary test. I'm going to merge it on red and test it on master. When all good, I'll backfill it to v1.9 and v1.10 manually.

**Pull request merged**